paasta_tools/kubernetes_tools.py

```diff
@@ -529,53 +575,106 @@ def format_kubernetes_app(self) -> V1Deployment:
             aws_ebs_volumes = self.get_aws_ebs_volumes()
 
             code_sha = get_code_sha_from_dockerurl(docker_url)
-            complete_config = V1Deployment(
-                api_version='apps/v1',
-                kind='Deployment',
-                metadata=V1ObjectMeta(
-                    name="{service}-{instance}".format(
-                        service=self.get_sanitised_service_name(),
-                        instance=self.get_sanitised_instance_name(),
+            if self.get_persistent_volumes():
+                complete_config = V1StatefulSet(
+                    api_version='apps/v1',
+                    kind='StatefulSet',
+                    metadata=V1ObjectMeta(
+                        name="{service}-{instance}".format(
```

Comment:
can you reduce duplication a bit? `name`, `service_name`,  `labels`, `containers`, `volumes`, `replicas` are (as far as i can tell) identical between these two branches.

Reply:
👍 good shout, let me take a crack